src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp

```diff
@@ -53,7 +53,14 @@ using org::eclipse::cyclonedds::topic::extensibility;
 using org::eclipse::cyclonedds::topic::encoding_version;
 using org::eclipse::cyclonedds::topic::TopicTraits;
 
-template<typename T>
+template<typename T, std::enable_if_t<TopicTraits<T>::isKeyless(), bool> = true >
+bool to_key(const T&, ddsi_keyhash_t& hash)
+{
+  memset(&(hash.value), 0x0, sizeof(hash.value));
+  return false;
```

Comment:
Doesn't this `false` imply that `populate_hash` still runs MD5 over it?

Reply:
It does, yes.

And with all instances having the same key value for a keyless topic, it should not matter whether this is hashed or not.